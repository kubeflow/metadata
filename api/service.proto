// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Placeholder for Core API definitions.

syntax = "proto3";

option go_package = "github.com/kubeflow/metadata/api";

package api;

import "api/api.proto";
import "google/api/annotations.proto";
import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "ml_metadata/proto/metadata_store.proto";

message CreateArtifactTypeRequest {
  ml_metadata.ArtifactType artifact_type = 1;
}

message CreateArtifactTypeResponse {
  // Newly created artifact type with id.
  ml_metadata.ArtifactType artifact_type = 1;
}

message GetArtifactTypeRequest {
  // ArtifactType names are of the form `artifact_types/{namespace}/{name}`.
  string name = 1;
}

message GetArtifactTypeResponse {
  ml_metadata.ArtifactType artifact_type = 1;
}

message ListArtifactTypesRequest {}

message ListArtifactTypesResponse {
  repeated ml_metadata.ArtifactType artifact_types = 1;
}

message DeleteArtifactTypeRequest {
  string name = 1;
}

message CreateArtifactRequest {
  // Creates the specified artifact as an instance of ArtifactType with this
  // fully qualified name. |parent| takes the form
  // `artifact_types/{namespace}/{name>}`.
  string parent = 1;
  // The Artifact to create. Note that Artifact.type_id is ignored.
  ml_metadata.Artifact artifact = 2;
}

message CreateArtifactResponse {
  // Newly created artifact with id.
  ml_metadata.Artifact artifact = 1;
}

message GetArtifactRequest {
  // Artifact name is like
  // `artifact_types/{namespace}/{typename}/artifact/{id}`.
  string name = 1;
}

message GetArtifactResponse {
  ml_metadata.Artifact artifact = 1;
}

message ListArtifactsRequest {
  string name = 1;
}

message ListArtifactsResponse {
  repeated ml_metadata.Artifact artifacts = 1;
}

message DeleteArtifactRequest {
  string name = 1;
}

message CreateExecutionTypeRequest {
  ml_metadata.ExecutionType execution_type = 1;
}

message CreateExecutionTypeResponse {
  // Newly created execution type with id.
  ml_metadata.ExecutionType execution_type = 1;
}

message GetExecutionTypeRequest {
  string name = 1;
}

message GetExecutionTypeResponse {
  ml_metadata.ExecutionType execution_type = 1;
}

message ListExecutionTypesRequest {}

message ListExecutionTypesResponse {
  repeated ml_metadata.ExecutionType execution_types = 1;
}

message DeleteExecutionTypeRequest {
  string name = 1;
}

message CreateExecutionRequest {
  // Creates the specified artifact as an instance of ExecutionType with this
  // fully qualified name. |parent| takes the form
  // `execution_types/{namespace}/{name>}`.
  string parent = 1;
  // The Execution to create. Note that Execution.type_id is ignored.
  ml_metadata.Execution execution = 2;
}

message CreateExecutionResponse {
  // Newly created execution with id.
  ml_metadata.Execution execution = 1;
}

message GetExecutionRequest {
  // Execution name is like
  // `execution_types/{namespace}/{typename}/execution/{id}`.
  string name = 1;
}

message GetExecutionResponse {
  ml_metadata.Execution execution = 1;
}

message ListExecutionsRequest {
  string name = 1;
}

message ListExecutionsResponse {
  repeated ml_metadata.Execution executions = 1;
}

message DeleteExecutionRequest {
  string name = 1;
}

message CreateEventRequest {
  ml_metadata.Event event = 1;
}

message SearchEventsRequest {
  string name = 1;
}

message SearchEventsResponse {
  repeated ml_metadata.Event events = 1;
}

service MetadataService {
  // NOTE:
  // The order of the following RPC methods affects the order of matching
  // a particular HTTP path. So put a more specific path pattern before
  // a generic one. For example,
  // GET /api/v1alpha1/{parent=artifact_types/**}/artifacts
  // should appear before
  // GET /api/v1alpha1/{name=artifact_types/**} to be possibly matched.
  rpc CreateArtifact(CreateArtifactRequest)
      returns (CreateArtifactResponse) {
    option (google.api.http) = {
      post: "/api/v1alpha1/{parent=artifact_types/**}/artifacts"
      body: "artifact"
    };
  }

  rpc GetArtifact(GetArtifactRequest)
      returns (GetArtifactResponse) {
    option (google.api.http) = {
      get: "/api/v1alpha1/{name=artifact_types/**/artifacts/*}"
    };
  }

  rpc ListArtifacts(ListArtifactsRequest)
      returns (ListArtifactsResponse) {
    option (google.api.http) = {
      // List artifacts for a given artifact type.
      get: "/api/v1alpha1/{name=artifact_types/**}/artifacts"
      additional_bindings {
        // List all artifacts.
        get: "/api/v1alpha1/artifacts"
      }
    };
  }

  rpc DeleteArtifact(DeleteArtifactRequest)
      returns (google.protobuf.Empty) {
    option (google.api.http) = {
      delete: "/api/v1alpha1/{name=artifact_types/**/artifacts/*}"
    };
  }

  rpc CreateExecution(CreateExecutionRequest)
      returns (CreateExecutionResponse) {
    option (google.api.http) = {
      post: "/api/v1alpha1/{parent=execution_types/**}/executions"
      body: "execution"
    };
  }

  rpc GetExecution(GetExecutionRequest)
      returns (GetExecutionResponse) {
    option (google.api.http) = {
      get: "/api/v1alpha1/{name=execution_types/**/executions/*}"
    };
  }

  rpc ListExecutions(ListExecutionsRequest)
      returns (ListExecutionsResponse) {
    option (google.api.http) = {
      get: "/api/v1alpha1/{name=execution_types/**}/executions"
      additional_bindings {
        // List all executions.
        get: "/api/v1alpha1/executions"
      }
    };
  }

  rpc DeleteExecution(DeleteExecutionRequest)
      returns (google.protobuf.Empty) {
    option (google.api.http) = {
      delete: "/api/v1alpha1/{name=execution_types/**/executions/*}"
    };
  }

  rpc CreateArtifactType(CreateArtifactTypeRequest)
      returns (CreateArtifactTypeResponse) {
    option (google.api.http) = {
      post: "/api/v1alpha1/artifact_types"
      body: "artifact_type"
    };
  }

  rpc ListArtifactTypes(ListArtifactTypesRequest)
      returns (ListArtifactTypesResponse) {
    option (google.api.http) = {
      get: "/api/v1alpha1/artifact_types"
    };
  }

  rpc GetArtifactType(GetArtifactTypeRequest)
      returns (GetArtifactTypeResponse) {
    option (google.api.http) = {
      get: "/api/v1alpha1/{name=artifact_types/**}"
    };
  }

  rpc DeleteArtifactType(DeleteArtifactTypeRequest)
      returns (google.protobuf.Empty) {
    option (google.api.http) = {
      delete: "/api/v1alpha1/{name=artifact_types/**}"
    };
  }


  rpc CreateExecutionType(CreateExecutionTypeRequest)
      returns (CreateExecutionTypeResponse) {
    option (google.api.http) = {
      post: "/api/v1alpha1/execution_types"
      body: "execution_type"
    };
  }

  rpc ListExecutionTypes(ListExecutionTypesRequest)
      returns (ListExecutionTypesResponse) {
    option (google.api.http) = {
      get: "/api/v1alpha1/execution_types"
    };
  }

  rpc GetExecutionType(GetExecutionTypeRequest)
      returns (GetExecutionTypeResponse) {
    option (google.api.http) = {
      get: "/api/v1alpha1/{name=execution_types/**}"
    };
  }

  rpc DeleteExecutionType(DeleteExecutionTypeRequest)
      returns (google.protobuf.Empty) {
    option (google.api.http) = {
      delete: "/api/v1alpha1/{name=execution_types/**}"
    };
  }

  rpc CreateEvent(CreateEventRequest)
      returns (google.protobuf.Empty) {
    option (google.api.http) = {
      post: "/api/v1alpha1/events"
      body: "event"
    };
  }

  // Search events based on an artifact or execution id.
  rpc SearchEvents(SearchEventsRequest)
      returns (SearchEventsResponse) {
    option (google.api.http) = {
      get: "/api/v1alpha1/events/{name=exections/*}:search"
      additional_bindings {
        get: "/api/v1alpha1/events/{name=artifacts/*}:search"
      }
    };
  }
}
