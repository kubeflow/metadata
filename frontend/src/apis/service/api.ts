/// <reference path="./custom.d.ts" />
// tslint:disable
/**
 * api/service.proto
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: version not set
 * 
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */


import * as url from "url";
import * as portableFetch from "portable-fetch";
import { Configuration } from "./configuration";

const BASE_PATH = "http://localhost".replace(/\/+$/, "");

/**
 *
 * @export
 */
export const COLLECTION_FORMATS = {
    csv: ",",
    ssv: " ",
    tsv: "\t",
    pipes: "|",
};

/**
 *
 * @export
 * @interface FetchAPI
 */
export interface FetchAPI {
    (url: string, init?: any): Promise<Response>;
}

/**
 *  
 * @export
 * @interface FetchArgs
 */
export interface FetchArgs {
    url: string;
    options: any;
}

/**
 * 
 * @export
 * @class BaseAPI
 */
export class BaseAPI {
    protected configuration: Configuration;

    constructor(configuration?: Configuration, protected basePath: string = BASE_PATH, protected fetch: FetchAPI = portableFetch) {
        if (configuration) {
            this.configuration = configuration;
            this.basePath = configuration.basePath || this.basePath;
        }
    }
};

/**
 * 
 * @export
 * @class RequiredError
 * @extends {Error}
 */
export class RequiredError extends Error {
    name: "RequiredError"
    constructor(public field: string, msg?: string) {
        super(msg);
    }
}

/**
 * ArtifactType represents the type of an ML artifact. It is used to define the schema of a named type.
 * @export
 * @interface ApiArtifactType
 */
export interface ApiArtifactType {
    /**
     * System-provided id.
     * @type {string}
     * @memberof ApiArtifactType
     */
    id?: string;
    /**
     * Required. Represents the name of a type. Must be unique within a namespace.
     * @type {string}
     * @memberof ApiArtifactType
     */
    name?: string;
    /**
     * Represents the namespace under which to store this type. Empty defaults to the default, system-defined namespace.
     * @type {ApiNamespace}
     * @memberof ApiArtifactType
     */
    namespace?: ApiNamespace;
    /**
     * Schema representing the fields and their types for this artifact.
     * @type {{ [key: string]: ApiType; }}
     * @memberof ApiArtifactType
     */
    type_properties?: { [key: string]: ApiType; };
    /**
     * An optional human-readable description of this type.
     * @type {string}
     * @memberof ApiArtifactType
     */
    description?: string;
}

/**
 * 
 * @export
 * @interface ApiCreateArtifactTypeResponse
 */
export interface ApiCreateArtifactTypeResponse {
    /**
     * Newly created artifact type with id.
     * @type {ApiArtifactType}
     * @memberof ApiCreateArtifactTypeResponse
     */
    artifact_type?: ApiArtifactType;
}

/**
 * Represents double types.
 * @export
 * @interface ApiDoubleType
 */
export interface ApiDoubleType {
    /**
     * An optional OpenAPI-compatible field-validation schema for this field.
     * @type {string}
     * @memberof ApiDoubleType
     */
    validator?: string;
}

/**
 * 
 * @export
 * @interface ApiGetArtifactTypeResponse
 */
export interface ApiGetArtifactTypeResponse {
    /**
     * 
     * @type {ApiArtifactType}
     * @memberof ApiGetArtifactTypeResponse
     */
    artifact_type?: ApiArtifactType;
}

/**
 * Represents integer types.
 * @export
 * @interface ApiIntType
 */
export interface ApiIntType {
    /**
     * An optional OpenAPI-compatible field-validation schema for this field.
     * @type {string}
     * @memberof ApiIntType
     */
    validator?: string;
}

/**
 * 
 * @export
 * @interface ApiListArtifactTypesResponse
 */
export interface ApiListArtifactTypesResponse {
    /**
     * 
     * @type {Array&lt;ApiArtifactType&gt;}
     * @memberof ApiListArtifactTypesResponse
     */
    artifact_types?: Array<ApiArtifactType>;
}

/**
 * Namespaces represent a named collection of ArtifactTypes and ExecutionTypes. This ensures multiple organizations can define ML schemas without type name collision.
 * @export
 * @interface ApiNamespace
 */
export interface ApiNamespace {
    /**
     * E.g. 'kubeflow.org/pipelines/v1' or 'myorganization.com/ml/v1'.
     * @type {string}
     * @memberof ApiNamespace
     */
    name?: string;
}

/**
 * Represents string types.
 * @export
 * @interface ApiStringType
 */
export interface ApiStringType {
    /**
     * An optional OpenAPI-compatible field-validation schema for this field.
     * @type {string}
     * @memberof ApiStringType
     */
    validator?: string;
}

/**
 * Type is the type of a field.
 * @export
 * @interface ApiType
 */
export interface ApiType {
    /**
     * 
     * @type {ApiStringType}
     * @memberof ApiType
     */
    string_type?: ApiStringType;
    /**
     * 
     * @type {ApiIntType}
     * @memberof ApiType
     */
    int_type?: ApiIntType;
    /**
     * 
     * @type {ApiDoubleType}
     * @memberof ApiType
     */
    double_type?: ApiDoubleType;
}


/**
 * MetadataServiceApi - fetch parameter creator
 * @export
 */
export const MetadataServiceApiFetchParamCreator = function (configuration?: Configuration) {
    return {
        /**
         * 
         * @param {ApiArtifactType} body 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createArtifactType(body: ApiArtifactType, options: any = {}): FetchArgs {
            // verify required parameter 'body' is not null or undefined
            if (body === null || body === undefined) {
                throw new RequiredError('body','Required parameter body was null or undefined when calling createArtifactType.');
            }
            const localVarPath = `/api/v1alpha1/artifact_types`;
            const localVarUrlObj = url.parse(localVarPath, true);
            const localVarRequestOptions = Object.assign({ method: 'POST' }, options);
            const localVarHeaderParameter = {} as any;
            const localVarQueryParameter = {} as any;

            localVarHeaderParameter['Content-Type'] = 'application/json';

            localVarUrlObj.query = Object.assign({}, localVarUrlObj.query, localVarQueryParameter, options.query);
            // fix override query string Detail: https://stackoverflow.com/a/7517673/1077943
            delete localVarUrlObj.search;
            localVarRequestOptions.headers = Object.assign({}, localVarHeaderParameter, options.headers);
            const needsSerialization = (<any>"ApiArtifactType" !== "string") || localVarRequestOptions.headers['Content-Type'] === 'application/json';
            localVarRequestOptions.body =  needsSerialization ? JSON.stringify(body || {}) : (body || "");

            return {
                url: url.format(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * 
         * @param {string} id 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteArtifactType(id: string, options: any = {}): FetchArgs {
            // verify required parameter 'id' is not null or undefined
            if (id === null || id === undefined) {
                throw new RequiredError('id','Required parameter id was null or undefined when calling deleteArtifactType.');
            }
            const localVarPath = `/api/v1alpha1/artifact_types/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            const localVarUrlObj = url.parse(localVarPath, true);
            const localVarRequestOptions = Object.assign({ method: 'DELETE' }, options);
            const localVarHeaderParameter = {} as any;
            const localVarQueryParameter = {} as any;

            localVarUrlObj.query = Object.assign({}, localVarUrlObj.query, localVarQueryParameter, options.query);
            // fix override query string Detail: https://stackoverflow.com/a/7517673/1077943
            delete localVarUrlObj.search;
            localVarRequestOptions.headers = Object.assign({}, localVarHeaderParameter, options.headers);

            return {
                url: url.format(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * 
         * @param {string} id 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getArtifactType(id: string, options: any = {}): FetchArgs {
            // verify required parameter 'id' is not null or undefined
            if (id === null || id === undefined) {
                throw new RequiredError('id','Required parameter id was null or undefined when calling getArtifactType.');
            }
            const localVarPath = `/api/v1alpha1/artifact_type/{id}`
                .replace(`{${"id"}}`, encodeURIComponent(String(id)));
            const localVarUrlObj = url.parse(localVarPath, true);
            const localVarRequestOptions = Object.assign({ method: 'GET' }, options);
            const localVarHeaderParameter = {} as any;
            const localVarQueryParameter = {} as any;

            localVarUrlObj.query = Object.assign({}, localVarUrlObj.query, localVarQueryParameter, options.query);
            // fix override query string Detail: https://stackoverflow.com/a/7517673/1077943
            delete localVarUrlObj.search;
            localVarRequestOptions.headers = Object.assign({}, localVarHeaderParameter, options.headers);

            return {
                url: url.format(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listArtifactTypes(options: any = {}): FetchArgs {
            const localVarPath = `/api/v1alpha1/artifact_types`;
            const localVarUrlObj = url.parse(localVarPath, true);
            const localVarRequestOptions = Object.assign({ method: 'GET' }, options);
            const localVarHeaderParameter = {} as any;
            const localVarQueryParameter = {} as any;

            localVarUrlObj.query = Object.assign({}, localVarUrlObj.query, localVarQueryParameter, options.query);
            // fix override query string Detail: https://stackoverflow.com/a/7517673/1077943
            delete localVarUrlObj.search;
            localVarRequestOptions.headers = Object.assign({}, localVarHeaderParameter, options.headers);

            return {
                url: url.format(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    }
};

/**
 * MetadataServiceApi - functional programming interface
 * @export
 */
export const MetadataServiceApiFp = function(configuration?: Configuration) {
    return {
        /**
         * 
         * @param {ApiArtifactType} body 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createArtifactType(body: ApiArtifactType, options?: any): (fetch?: FetchAPI, basePath?: string) => Promise<ApiCreateArtifactTypeResponse> {
            const localVarFetchArgs = MetadataServiceApiFetchParamCreator(configuration).createArtifactType(body, options);
            return (fetch: FetchAPI = portableFetch, basePath: string = BASE_PATH) => {
                return fetch(basePath + localVarFetchArgs.url, localVarFetchArgs.options).then((response) => {
                    if (response.status >= 200 && response.status < 300) {
                        return response.json();
                    } else {
                        throw response;
                    }
                });
            };
        },
        /**
         * 
         * @param {string} id 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteArtifactType(id: string, options?: any): (fetch?: FetchAPI, basePath?: string) => Promise<any> {
            const localVarFetchArgs = MetadataServiceApiFetchParamCreator(configuration).deleteArtifactType(id, options);
            return (fetch: FetchAPI = portableFetch, basePath: string = BASE_PATH) => {
                return fetch(basePath + localVarFetchArgs.url, localVarFetchArgs.options).then((response) => {
                    if (response.status >= 200 && response.status < 300) {
                        return response.json();
                    } else {
                        throw response;
                    }
                });
            };
        },
        /**
         * 
         * @param {string} id 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getArtifactType(id: string, options?: any): (fetch?: FetchAPI, basePath?: string) => Promise<ApiGetArtifactTypeResponse> {
            const localVarFetchArgs = MetadataServiceApiFetchParamCreator(configuration).getArtifactType(id, options);
            return (fetch: FetchAPI = portableFetch, basePath: string = BASE_PATH) => {
                return fetch(basePath + localVarFetchArgs.url, localVarFetchArgs.options).then((response) => {
                    if (response.status >= 200 && response.status < 300) {
                        return response.json();
                    } else {
                        throw response;
                    }
                });
            };
        },
        /**
         * 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listArtifactTypes(options?: any): (fetch?: FetchAPI, basePath?: string) => Promise<ApiListArtifactTypesResponse> {
            const localVarFetchArgs = MetadataServiceApiFetchParamCreator(configuration).listArtifactTypes(options);
            return (fetch: FetchAPI = portableFetch, basePath: string = BASE_PATH) => {
                return fetch(basePath + localVarFetchArgs.url, localVarFetchArgs.options).then((response) => {
                    if (response.status >= 200 && response.status < 300) {
                        return response.json();
                    } else {
                        throw response;
                    }
                });
            };
        },
    }
};

/**
 * MetadataServiceApi - factory interface
 * @export
 */
export const MetadataServiceApiFactory = function (configuration?: Configuration, fetch?: FetchAPI, basePath?: string) {
    return {
        /**
         * 
         * @param {ApiArtifactType} body 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        createArtifactType(body: ApiArtifactType, options?: any) {
            return MetadataServiceApiFp(configuration).createArtifactType(body, options)(fetch, basePath);
        },
        /**
         * 
         * @param {string} id 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        deleteArtifactType(id: string, options?: any) {
            return MetadataServiceApiFp(configuration).deleteArtifactType(id, options)(fetch, basePath);
        },
        /**
         * 
         * @param {string} id 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        getArtifactType(id: string, options?: any) {
            return MetadataServiceApiFp(configuration).getArtifactType(id, options)(fetch, basePath);
        },
        /**
         * 
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        listArtifactTypes(options?: any) {
            return MetadataServiceApiFp(configuration).listArtifactTypes(options)(fetch, basePath);
        },
    };
};

/**
 * MetadataServiceApi - object-oriented interface
 * @export
 * @class MetadataServiceApi
 * @extends {BaseAPI}
 */
export class MetadataServiceApi extends BaseAPI {
    /**
     * 
     * @param {} body 
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MetadataServiceApi
     */
    public createArtifactType(body: ApiArtifactType, options?: any) {
        return MetadataServiceApiFp(this.configuration).createArtifactType(body, options)(this.fetch, this.basePath);
    }

    /**
     * 
     * @param {} id 
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MetadataServiceApi
     */
    public deleteArtifactType(id: string, options?: any) {
        return MetadataServiceApiFp(this.configuration).deleteArtifactType(id, options)(this.fetch, this.basePath);
    }

    /**
     * 
     * @param {} id 
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MetadataServiceApi
     */
    public getArtifactType(id: string, options?: any) {
        return MetadataServiceApiFp(this.configuration).getArtifactType(id, options)(this.fetch, this.basePath);
    }

    /**
     * 
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof MetadataServiceApi
     */
    public listArtifactTypes(options?: any) {
        return MetadataServiceApiFp(this.configuration).listArtifactTypes(options)(this.fetch, this.basePath);
    }

}

