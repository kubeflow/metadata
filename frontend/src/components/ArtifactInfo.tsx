/*
 * Copyright 2019 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from 'react';
import {stylesheet} from 'typestyle';
import {color, commonCss} from '../Css';
import {formatDateString, getArtifactProperty} from '../lib/Utils';
import {MlMetadataArtifact} from '../apis/service';
import {ArtifactProperties, ArtifactCustomProperties} from '../lib/Api';
import {ModelInfo} from './ModelInfo';

export const css = stylesheet({
  artifactInfo: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  field: {
    flexBasis: '300px',
    marginBottom: '32px',
  },
  term: {
    color: color.grey,
    fontSize: '12px',
    letterSpacing: '0.2px',
    lineHeight: '16px',
  },
  value: {
    color: color.secondaryText,
    fontSize: '14px',
    letterSpacing: '0.2px',
    lineHeight: '20px',
  }
});

export interface ArtifactInfoProps {
  typeName: string;
  artifact: MlMetadataArtifact;
}

export class ArtifactInfo extends React.Component<ArtifactInfoProps, {}> {

  public render(): JSX.Element {
    const {artifact} = this.props;
    const createTime = formatDateString(
      getArtifactProperty(artifact, ArtifactProperties.CREATE_TIME) || '');
    return (
      <section>
        <h1 className={commonCss.header}>{this.props.typeName} info</h1>
        <dl className={css.artifactInfo}>
          <div className={css.field}>
            <dt className={css.term}>Version ID</dt>
            <dd className={css.value}>
              {getArtifactProperty(artifact, ArtifactProperties.VERSION)}
            </dd>
          </div>
          <div className={css.field}>
            <dt className={css.term}>Workspace</dt>
            <dd className={css.value}>
              {getArtifactProperty(artifact, ArtifactCustomProperties.WORKSPACE,
                true)}
            </dd>
          </div>
          <div className={css.field}>
            <dt className={css.term}>Run</dt>
            <dd className={css.value}>
              {getArtifactProperty(artifact, ArtifactCustomProperties.RUN,
                true)}
            </dd>
          </div>
          <div className={css.field}>
            <dt className={css.term}>Description</dt>
            <dd className={css.value}>
              {getArtifactProperty(artifact, ArtifactProperties.DESCRIPTION)}
            </dd>
          </div>
          <div className={css.field}>
            <dt className={css.term}>Creation time</dt>
            <dd className={css.value}>{createTime}</dd>
          </div>
          {this.props.typeName === 'Model' ?
            <ModelInfo model={artifact} /> :
            <div className={css.field}>
              <dt className={css.term}>URI</dt>
              <dd className={css.value}>{artifact.uri}</dd>
            </div>
          }
        </dl>
      </section>
    );
  }
}
