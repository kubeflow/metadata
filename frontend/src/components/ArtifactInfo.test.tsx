import * as React from 'react';
import {ArtifactInfo} from './ArtifactInfo';
import {shallow, ShallowWrapper} from 'enzyme';
import {MlMetadataArtifact} from '../apis/service';

describe('ArtifactInfo', () => {
  let tree: ShallowWrapper;

  afterEach(() => tree.unmount());

  it('Renders generic Artifact information', () => {
    const dataset: MlMetadataArtifact = {
      id: '2',
      type_id: '3',
      uri: 'gs://my-bucket/dataset',
      properties: {
        name: {string_value: 'test dataset'},
        version: {string_value: 'v0.0.1'},
        create_time: {string_value: '2019-06-12T01:21:48.259263Z'},
      },
      custom_properties: {
        __kf_workspace__: {string_value: 'workspace-1'},
      },
    };
    tree = shallow(<ArtifactInfo typeName='Data Set' artifact={dataset} />);

    expect(tree).toMatchSnapshot();
  });

  it('Renders Artifact information for a Model', () => {
    const model: MlMetadataArtifact = {
      id: '1',
      type_id: '1',
      uri: 'gs://my-bucket/mnist',
      properties: {
        name: {string_value: 'test model'},
        description: {string_value: 'A really great model'},
        version: {string_value: 'v1'},
        create_time: {string_value: '2019-06-12T01:21:48.259263Z'},
        __ALL_META__: {
          string_value: '{"hyperparameters": {"early_stop": true, ' +
            '"layers": [10, 3, 1], "learning_rate": 0.5}, ' +
            '"model_type": "neural network", ' +
            '"training_framework": {"name": "tensorflow", "version": "v1.0"}}'
        }
      },
      custom_properties: {
        __kf_workspace__: {string_value: 'workspace-1'},
        __kf_run__: {string_value: '1'},
      },
    };
    tree = shallow(<ArtifactInfo typeName='Model' artifact={model} />);

    expect(tree).toMatchSnapshot();
  });
});
