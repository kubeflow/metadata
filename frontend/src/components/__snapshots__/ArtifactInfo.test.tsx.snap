// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtifactInfo Renders Artifact information for a Model 1`] = `
<section>
  <h1
    className="header_f189n3y"
  >
    Model
     info
  </h1>
  <dl
    className="artifactInfo_fnuket9"
  >
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Version ID
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        v1
      </dd>
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Workspace
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        workspace-1
      </dd>
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Run
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        1
      </dd>
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Description
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        A really great model
      </dd>
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Creation time
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        6/11/2019, 9:21:48 PM
      </dd>
    </div>
    <ModelInfo
      model={
        Object {
          "custom_properties": Object {
            "__kf_run__": Object {
              "string_value": "1",
            },
            "__kf_workspace__": Object {
              "string_value": "workspace-1",
            },
          },
          "id": "1",
          "properties": Object {
            "__ALL_META__": Object {
              "string_value": "{\\"hyperparameters\\": {\\"early_stop\\": true, \\"layers\\": [10, 3, 1], \\"learning_rate\\": 0.5}, \\"model_type\\": \\"neural network\\", \\"training_framework\\": {\\"name\\": \\"tensorflow\\", \\"version\\": \\"v1.0\\"}}",
            },
            "create_time": Object {
              "string_value": "2019-06-12T01:21:48.259263Z",
            },
            "description": Object {
              "string_value": "A really great model",
            },
            "name": Object {
              "string_value": "test model",
            },
            "version": Object {
              "string_value": "v1",
            },
          },
          "type_id": "1",
          "uri": "gs://my-bucket/mnist",
        }
      }
    />
  </dl>
</section>
`;

exports[`ArtifactInfo Renders generic Artifact information 1`] = `
<section>
  <h1
    className="header_f189n3y"
  >
    Data Set
     info
  </h1>
  <dl
    className="artifactInfo_fnuket9"
  >
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Version ID
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        v0.0.1
      </dd>
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Workspace
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        workspace-1
      </dd>
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Run
      </dt>
      <dd
        className="value_f1d9lp3k"
      />
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Description
      </dt>
      <dd
        className="value_f1d9lp3k"
      />
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        Creation time
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        6/11/2019, 9:21:48 PM
      </dd>
    </div>
    <div
      className="field_f1bgv1eo"
    >
      <dt
        className="term_fuqr2el"
      >
        URI
      </dt>
      <dd
        className="value_f1d9lp3k"
      >
        gs://my-bucket/dataset
      </dd>
    </div>
  </dl>
</section>
`;
